lib/uv.sh

```diff
@@ -0,0 +1,165 @@
+
+	# The earlier buildpack cache invalidation step will have already handled the case where
+	# the uv version has changed, so here we only need to check whether the uv binary exists.
+	if [[ -f "${uv_dir}/uv" ]]; then
+		output::step "Using cached uv ${UV_VERSION}"
+	else
+		output::step "Installing uv ${UV_VERSION}"
+		mkdir -p "${uv_dir}"
+
+		local gnu_arch
+		# eg: `x86_64` or `aarch64`.
+		gnu_arch=$(arch)
+		local uv_url="https://github.com/astral-sh/uv/releases/download/${UV_VERSION}/uv-${gnu_arch}-unknown-linux-gnu.tar.gz"
+
+		if ! {
+			# We set max-time for improved UX for hanging downloads compared to relying on the build system
+			# timeout. The uv archive is only ~15 MB so takes < 1s to download on Heroku's build system,
+			# however, we use much higher timeouts so that the buildpack works in non-Heroku environments
+			# that may be far from `us-east-1` or have a slower connection. We don't use `--speed-limit`
+			# since it gives worse error messages when used with retries and piping to tar.
+			curl \
+				--connect-timeout 10 \
+				--fail \
+				--location \
+				--max-time 120 \
+				--retry-max-time 120 \
+				--retry 3 \
+				--retry-connrefused \
+				--show-error \
+				--silent \
+				"${uv_url}" \
+				| tar \
+					--directory "${uv_dir}" \
+					--extract \
+					--gzip \
+					--no-anchored \
+					--strip-components 1 \
+					uv
+		}; then
+			output::error <<-EOF
+				Error: Unable to install uv.
+
+				Failed to download/install uv from GitHub:
+				${uv_url}
+
+				In some cases, this happens due to a temporary issue with
+				the network connection or GitHub's API/CDN.
+
+				Try building again to see if the error resolves itself.
+
+				If that doesn't help, check the status of GitHub here:
+				https://www.githubstatus.com
+			EOF
+			meta_set "failure_reason" "install-package-manager::uv"
+			exit 1
+		fi
+	fi
+
+	export PATH="${uv_dir}:${PATH}"
+	# Make uv manage the system site-packages of our Python install instead of creating a venv.
+	export UV_PROJECT_ENVIRONMENT="${python_home}"
+	# Prevent uv from downloading/using its own Python installation.
+	export UV_PYTHON_DOWNLOADS="never"
+	export UV_PYTHON_PREFERENCE="only-system"
+
+	# Set the same env vars in the environment used by later buildpacks.
+	cat >>"${export_file}" <<-EOF
+		export PATH="${uv_dir}:\${PATH}"
+		export UV_PROJECT_ENVIRONMENT="${python_home}"
+		export UV_PYTHON_DOWNLOADS="never"
+		export UV_PYTHON_PREFERENCE="only-system"
+	EOF
+
+	# As a performance optimisation, uv attempts to use hardlinks instead of copying files from its
+	# download cache into site-packages, and will emit a warning if it has to fall back to copying.
+	# By default uv stores its cache under `$HOME/.cache`, and for standard Heroku builds `$HOME` is
+	# `/app`, which is on a different filesystem mount to the build directory (which is under `/tmp`),
+	# meaning hardlinks can't be used. To avoid this we tell uv's to store its cache in `/tmp`.
```

Comment:
```suggestion
	# meaning hardlinks can't be used. To avoid this we tell uv to store its cache in `/tmp`.
```

Reply:
Thank you - typo fixed in f8470f6b536172216f567f1cfe43f15980ec58e9